src/mccode_antlr/translators/c_initialise.py

```diff
@@ -244,6 +248,18 @@ def cogen_initialize(source, component_declared_parameters, ok_to_skip):
     lines.extend([
         f'int init(void) {{ /* called by mccode_main for {source.name}:INITIALIZE */',
         '  DEBUG_INSTR();',
+    ])
+    if version > Version("3.5.40"):
+        # McCode https://github.com/mccode-dev/McCode/pull/2227#issuecomment-3613096573
+        # changed `mccode_main.c` and the code-generator to move the random number
+        # seeding before "INITIALIZE" time.
+        # The most-recent release at that time was 3.5.40, so this change is only
+        # compatible with releases after that.
+        lines.extend([
+            '// Initialise rng',
+            '  srandom(_hash(mcseed-1));',
+        ])
```

Comment:
@willend Is my assumption here correct (that we only want this line in the code generator if we're using McCode `v3.5.41` or greater)?